uaclient/entitlements/base.py

```diff
@@ -455,6 +471,59 @@ def enable(
 
         return True, None
 
+    def handle_additional_snaps(self) -> bool:
+        """ "install snaps necessary to enable a service."""
+        additional_snaps = (
+            self.entitlement_cfg.get("entitlement", {})
+            .get("directives", {})
+            .get("additionalSnaps")
+        )
+
+        # If we don't have the directive, there is nothing
+        # to process here
+        if additional_snaps is None:
+            return True
+
+        if not system.which(snap.SNAP_CMD):
```

Comment:
Why not `if not snap.is_installed()`?

Reply:
Right now, we have two checks regarding snapd:

```python
if not system.which(snap.SNAP_CMD):
    pass
elif not snap.is_snapd_installed():
    pass
```

I think what this check is covering is the scenario where we have a `snap` command, but for some reason we don't have the `snapd` package installed. This is the logic we use today on Livepatch, and honestly, I don't know if we need it.

Maybe we can directly check if `snapd` is installed and if not, try to install it. What do you think ?

Reply:
Yeah I'm not sure why we need to check both - maybe @blackboxsw might remember something?

Reply:
@blackboxsw just double checking if you can verify this part of the code

Reply:
I don't recall why we check both here, my original comment on an earlier iteration of the same PR that introduced this "def is_installed" was basically [do we need both](https://github.com/canonical/ubuntu-pro-client/pull/1678#discussion_r660014208)   .... that said, there was a related PR that changed our subp(["snap"...]) to subp(["/usr/bin/snap",...])  in https://github.com/canonical/ubuntu-pro-client/commit/698b3ca5299dc3973f3bdea4fd58cca884effeac. so I don't think we even need the which anymore